src/evict/evict_lru.c

```diff
@@ -2904,13 +2890,18 @@ __wti_evict_app_assist_worker(WT_SESSION_IMPL *session, bool busy, bool readonly
         if (!busy && __wt_atomic_loadv64(&txn_shared->pinned_id) != WT_TXN_NONE &&
           __wt_atomic_loadv64(&txn_global->current) != __wt_atomic_loadv64(&txn_global->oldest_id))
             busy = true;
-        max_progress = busy ? 5 : 20;
+        uint64_t max_progress = busy ? 5 : 20;
 
         /* See if eviction is still needed. */
         if (!__wt_evict_needed(session, busy, readonly, &pct_full) ||
           (pct_full < 100.0 &&
             (__wt_atomic_loadv64(&evict->eviction_progress) > initial_progress + max_progress)))
-            break;
+            goto err;
+
+        if (!__evict_check_user_ok_with_eviction(session, busy)) {
+            ret = 0; /* This is a no-error exit. */
```

Comment:
Do we need to explicitly set this to 0, can't we just goto err; directly here?

Reply:
No, because by this point, all prior operations have either succeeded or any errors have already been handled. A user canceling further eviction is not an error condition - it should be treated as a normal termination of the loop. This ensures there’s no unintended error code leakage.

Reply:
Talked offline with Sid and Jie, `ret` gets reset to 0 at the end of each loop in the `for` statement. At this point it can only be 0, so it doesn't need to be assigned to 0.